### PR TITLE
add fsanitizer for address and undefined

### PR DESCRIPTION
**Pull request by mohsensadr**

### CODE REVIEW

**Comment by mohsensadr** [comment]
@aaadelmann  How about now?  [c8cd6ac](https://github.com/IPPL-framework/ippl/pull/242/commits/c8cd6accb6ca6c7fa7f7e3caa4adddc0e677e185)

**Review by aaadelmann — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -37,7 +37,7 @@ set (CMAKE_CXX_STANDARD 20)
 set (CMAKE_CXX_STANDARD_REQUIRED ON)
 set (CMAKE_CXX_FLAGS_RELWITHDEBINFO "-O3 -g ")
 set (CMAKE_CXX_FLAGS_RELEASE "-O3")
-set (CMAKE_CXX_FLAGS_DEBUG  "-O0 -g")
+set (CMAKE_CXX_FLAGS_DEBUG  "-O0 -g -fsanitize=address,undefined")
```
[the remark this answers is not in the document]
Comment (manuel5975p) [reply CMakeLists.txt]: Intel's ICX supports it (seen [here](https://godbolt.org/z/Ezn44bMGE)), so do clang and gcc, and our configuration is not msvc compliant as-is.

[the remark this answers is not in the document]
Comment (aaadelmann) [reply CMakeLists.txt]: msvc is not a propority

**Review by matt-frey — APPROVED** [review]
Looks good to me. Thanks.

**Pull request merged**